service/common/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -2040,7 +2034,13 @@ private List<TableIdentifier> listTableLike(PolarisEntitySubType subType, Namesp
    */
   private FileIO loadFileIO(String ioImpl, Map<String, String> properties) {
     Map<String, String> propertiesWithS3CustomizedClientFactory = new HashMap<>(properties);
-    return fileIOFactory.loadFileIO(ioImpl, propertiesWithS3CustomizedClientFactory);
+    return fileIOFactory.loadFileIO(
+        ioImpl,
+        propertiesWithS3CustomizedClientFactory,
+        null /*tableIdentifier*/,
+        null /*tableLocations*/,
+        null /*storageActions*/,
+        null /*resolvedStorageEntity*/);
```

Comment:
`null` looks a bit dangerous to me. Maybe add a `loadFileIO(ioImpl,properties)` and let the implementation do the "right" thing?